### ISSUE TITLE
TURN server limit

### ISSUE THREAD

**Issue by tenge-amzn** [coturn/coturn#616]
Hi,
I'm wondering if anyone has previously dug into this, what allocation/bandwidth/pps limit coturn can take?
I know this depends on the server specifications, but just for software perspective, do we know any limitation?

We've seen some packet loss when the pps approaching to a certain number, no matter what server size we use

**Comment by misi** [comment]
@tenge-amzn  I don't have such information.. How you measured packet loss?

**Comment by tenge-amzn** [comment]
In /proc/net/snmp, the UDP "RcvbufErrors" skyrocketed when load goes up, around 100pps or 700-800Mbps.

**Comment by misi** [comment]
So this mean I guess recv buffer overflow! Is any proxy before your service? It could cause problem in load-balance if source IP:port is the same, because it will loablanced to the same thread/core..

**Comment by tenge-amzn** [comment]
> So this mean I guess recv buffer overflow! Is any proxy before your service? It could cause problem in load-balance if source IP:port is the same, because it will loablanced to the same thread/core..

No. TURN servers are public facing and load balanced by DNS. The netengine is 3 - udp socket per thread. Also tried to increase udp socket buffer size, but only limited help.

**Comment by JalySN** [comment]
When the cpu of coturn exceeds 85%, many udp packets are lost
